backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java

```diff
@@ -274,4 +280,39 @@ public FutureExperimentRunDAO(Executor executor, FutureJdbi jdbi, UAC uac) {
         },
         executor);
   }
+
+  public InternalFuture<Void> logArtifacts(LogArtifacts request) {
+    final var runId = request.getId();
+    final var artifacts = request.getArtifactsList();
+    final var now = Calendar.getInstance().getTimeInMillis();
+
+    return checkPermission(runId, ModelDBActionEnum.ModelDBServiceActions.UPDATE)
+        .thenCompose(unused -> artifactHandler.logArtifacts(runId, artifacts), executor)
+        .thenCompose(unused -> updateModifiedTimestamp(runId, now), executor);
+  }
+
+  public InternalFuture<List<Artifact>> getArtifacts(GetArtifacts request) {
+    final var runId = request.getId();
+    final var key = request.getKey();
```

Comment:
```
message GetArtifacts {
    string id = 1;
    string key = 2;

    message Response {
        repeated common.Artifact artifacts = 1;
    }
}
```
So i have take key as optional. If only id is there then it will returns all artifact of that run and if key is present then matching key artifacts will be return.

Reply:
I mean that `request.getKey()` will always return a valid string. That's protobuf's internal implementation

Reply:
Ok, Updating it without Optional

Reply:
Removed optional for key

Reply:
That's not what I meant. Fixing

Reply:
Optional was the right choice. But its use was incorrect.